numbat/modules/core/lists.nbt

```diff
@@ -87,7 +87,7 @@ fn filter<A>(p: Fn[(A) -> Bool], xs: List<A>) -> List<A> =
       else filter(p, tail(xs))
 
 @description("Fold a function over a list")
-@example("foldl(str_append, \"\", [\"Num\", \"bat\", \"!\"])", "Join a list of strings by folding.")
+@example("foldl(str_prepend, \"\", [\"Num\", \"bat\", \"!\"])", "Join a list of strings by folding.")
```

Comment:
Well this change is… strange. The order reversal for `str_append` is consistent with the rest, sure. But it's also very much not the order I would expect. Can we maybe do some research how other functional programming languages handle this?

Reply:
Sure, I will start to look into it.

Reply:
I had a difficult time finding languages with both an `|>` operator equivalent and the append/prepend functions. Below are my findings so far. It seems to me that there isn't a consensus yet, but it is slightly more normal to have the arguments the way they originally were as opposed to the ones proposed in this PR. 

The same as proposed:
- https://hackage.haskell.org/package/strings-1.1/docs/Data-Strings.html

The same as original:
- https://www.roc-lang.org/builtins/Str#concat (under the name of concat, but the idea is the same)
- https://package.elm-lang.org/packages/elm/core/latest/String#append (only has append)

No 1-to1 comparison:
- https://hexdocs.pm/elixir/String.html#replace_prefix/3 (close to the original but via a different function)
- https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-stringmodule.html (uses the String class method and not a functional approach)
- https://share.unison-lang.org/@unison/base/code/releases/3.23.0/latest/terms/Text/cons (uses `cons` and `snoc` instead)

I don't really mind either order for the arguments and only changed it in this PR to stay consistent. I think either piping or normal function call will always feel weird. Maybe by changing the naming we can make it less so, but I'm unsure how that would look like.

Reply:
Having thought about it a bit since, I think the best solution might be to reverse the arguments such that `str_append` functions as it did before. To use them nicely in pipelines would require remembering that quirk, but that would be a lot less complicated when lambdas are added someday.

I think the pipelines are important, and they are what I use most often, but thinking of how beginners would perceive the function has lead me to believe that the confusion my change to the function would cause isn't worth it. 

If you agree, I will change the signature of `str_append` and `str_prepend` and update the brach to make sure it won't break any code merged in the meantime.

Reply:
> If you agree, I will change the signature of `str_append` and `str_prepend` and update the brach to make sure it won't break any code merged in the meantime.

That would be great, thanks.